Sources/MapboxNavigation/CarPlayNavigationViewController.swift

```diff
@@ -695,6 +696,42 @@ open class CarPlayNavigationViewController: UIViewController, BuildingHighlighti
         navigationMapView?.showWaypoints(on: progress.route, legIndex: legIndex)
     }
     
+    func updateLegacyIcons() {
```

Comment:
When `traitCollection` change or style change, clearing the cache of `spriteRepository` to remove the previously stored generic view and exit views. And then re-generating the `attributedText` for  `CPManeuver`.

Reply:
Why do we need to clear cache whenever style changes?

Reply:
This change now also leads to a problem when `CPManeuver` is updated more than once whenever CarPlay app goes to background and then to foreground. Similar with startup, when now update it three times in a row.

Reply:
I can also see that we call `updateManeuvers()` right in the `viewDidLoad()` of `CarPlayNavigationViewController`, shouldn't we call it only when instruction gets updated? Ideally we should update `CPManeuver` only once during initial launch.